source/common/buffer/buffer_impl.h

```diff
@@ -383,6 +422,10 @@ class Slice {
 
   /** Hooks to execute when the slice is destroyed. */
   std::list<std::function<void()>> drain_trackers_;
+
+  /** Charges associated with this slice. This should be 0 or 1 elements. When
+   * coalescing with another slice, we do not transfer over their charges. */
+  absl::InlinedVector<std::shared_ptr<Account>, 1> charges_;
```

Comment:
Since this can only be 0 or 1, consider changing this member to:
std::shared_ptr<Account> account_;